src/Type/Definition/NonNull.php

```diff
@@ -4,65 +4,24 @@
 
 namespace GraphQL\Type\Definition;
 
-use GraphQL\Utils\Utils;
-
 class NonNull extends Type implements WrappingType, OutputType, InputType
 {
-    /** @var NullableType */
+    /** @var Type&NullableType */
     private $ofType;
 
-    /**
-     * @param NullableType $type
-     */
-    public function __construct($type)
+    public function __construct(NullableType $type)
     {
-        $this->ofType = self::assertNullableType($type);
-    }
-
-    /**
-     * @param mixed $type
-     *
-     * @return NullableType
-     */
-    public static function assertNullableType($type)
-    {
-        Utils::invariant(
-            Type::isType($type) && ! $type instanceof self,
-            'Expected ' . Utils::printSafe($type) . ' to be a GraphQL nullable type.'
-        );
-
-        return $type;
-    }
-
-    /**
-     * @param mixed $type
-     *
-     * @return self
-     */
-    public static function assertNullType($type)
-    {
-        Utils::invariant(
-            $type instanceof self,
-            'Expected ' . Utils::printSafe($type) . ' to be a GraphQL Non-Null type.'
-        );
-
-        return $type;
+        /** @var Type&NullableType $nullableType*/
```

Comment:
This is a hack to get around a parsing bug in `phpcode_sniffer`--it can't handle the "Type&NullableType" form in the docblock, so I have to "cast" it here to appease `PHPStan`.

Reply:
@simPod I forked `PHP_CodeSniffer`(https://github.com/squizlabs/PHP_CodeSniffer) so I can debug this and possibly open a PR with a fix, but I'll be damned if I can figure out how to run tests for it from within PHPStorm (the `contributor.md` recommends running a global `phpunit` command from shell). Any ideas?

Reply:
Yup, I have it forked as well

Set test framework (don't forget the PHP interpreter as well)

![image](https://user-images.githubusercontent.com/327717/67881921-b1721300-fb41-11e9-9880-c63c1d0e6c18.png)

phpunit is dev dependency there, so no need for global.

Then run tests and youre good to go 
![image](https://user-images.githubusercontent.com/327717/67881983-c8186a00-fb41-11e9-9827-9e828e037e48.png)

Reply:
Ah, I must have missed the config file; I'll try it after work. Great, thanks so much!

Reply:
I wasn't able to come up with a quick fix for this, but I'm going to keep working on it. In the meantime, hopefully it won't hold up this PR.